### PR TITLE
feat: Add appsync arguments to support appsync security features

### PR DESCRIPTION
**Pull request by jagoodhand**
## Description
Add support for the new AppSync GraphQL controls for query depth, run complexity and introspection.

## Motivation and Context
This enables users to utilise the new AppSync features that improve the security of their AppSync APIs.

## Breaking Changes
This increases the minimum AWS provider version to 5.37, as this was the terraform provider that implemented these controls.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by jagoodhand** [comment]
@antonbabenko could I request your review on this PR please?

**Comment by jagoodhand** [comment]
@antonbabenko - is there anything else you would need me to do on this PR before it can be merged? Thanks.

File: variables.tf [hunk 1]
```diff
@@ -321,3 +321,35 @@ variable "functions" {
   type        = any
   default     = {}
 }
+variable "introspection_config" {
+  description = "Whether to enable or disable introspection of the GraphQL API."
+  type        = string
+  default     = "ENABLED"
```
Comment (antonbabenko) [inline variables.tf]: Please change the default value for all 3 new variables to `null` and remove `validation` section entirely from all variables.

**Review by antonbabenko — CHANGES_REQUESTED** [review]

**Comment by jagoodhand** [comment]
@antonbabenko - done. Would you mind re-reviewing please?

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 2.5.0 :tada:

**Pull request merged**